src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -118,39 +118,44 @@ private static void CascadeToPrimaryConstructorParameters(IPropertySymbol proper
     private static bool IsForEachProperty(IPropertySymbol symbol)
         => symbol.Name == WellKnownMemberNames.CurrentPropertyName;
 
-    protected sealed override async ValueTask<ImmutableArray<FinderLocation>> FindReferencesInDocumentAsync(
+    protected sealed override async ValueTask FindReferencesInDocumentAsync<TData>(
         IPropertySymbol symbol,
         FindReferencesDocumentState state,
+        Action<FinderLocation, TData> processResult,
+        TData processResultData,
         FindReferencesSearchOptions options,
         CancellationToken cancellationToken)
     {
-        var nameReferences = await FindReferencesInDocumentUsingSymbolNameAsync(
-            symbol, state, cancellationToken).ConfigureAwait(false);
-
-        if (options.AssociatePropertyReferencesWithSpecificAccessor)
-        {
-            // We want to associate property references to a specific accessor (if an accessor
-            // is being referenced).  Check if this reference would match an accessor. If so, do
-            // not add it.  It will be added by PropertyAccessorSymbolReferenceFinder.
-            nameReferences = nameReferences.WhereAsArray(loc =>
+        await FindReferencesInDocumentUsingSymbolNameAsync(
+            symbol,
+            state,
+            static (loc, data) =>
             {
-                var accessors = GetReferencedAccessorSymbols(
-                    state, symbol, loc.Node, cancellationToken);
-                return accessors.IsEmpty;
-            });
-        }
+                var useResult = true;
+                if (data.options.AssociatePropertyReferencesWithSpecificAccessor)
+                {
+                    // We want to associate property references to a specific accessor (if an accessor
+                    // is being referenced).  Check if this reference would match an accessor. If so, do
+                    // not add it.  It will be added by PropertyAccessorSymbolReferenceFinder.
+                    var accessors = GetReferencedAccessorSymbols(
+                        data.state, data.symbol, loc.Node, data.cancellationToken);
+                    useResult = accessors.IsEmpty;
+                }
+
+                if (useResult)
+                    data.processResult(loc, data.processResultData);
+            },
+            processResultData: (options, state, symbol, cancellationToken, self: this, processResult, processResultData),
```

Comment:
can you put 'this' at the front.

Reply:
and cancellationToken at the end?  keep the rest as much in order as the signature of the containing method.